server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -149,7 +150,8 @@ public void sendCanMatch(
         SearchTask task,
         final ActionListener<CanMatchNodeResponse> listener
     ) {
-        if (connection.getVersion().onOrAfter(Version.V_7_16_0) && connection.getNode().getVersion().onOrAfter(Version.V_7_16_0)) {
+        if (connection.getTransportVersion().onOrAfter(TransportVersion.V_7_16_0)
+            && connection.getNode().getVersion().onOrAfter(Version.V_7_16_0)) {
```

Comment:
I wonder what will getNode().getVersion return on serverless?

Reply:
It'll just return 8.7.0 for now, same as on-prem installs of the same codebase

Reply:
agree - let's keep this like this in this PR
if we are to refactor discoveryNode.getVersion to return transport version, it should be done in a separate PR